src/plugins/dashboard/public/dashboard_app/_dashboard_app_strings.ts

```diff
@@ -154,6 +154,21 @@ export const shareModalStrings = {
       defaultMessage:
         'One or more panels on this dashboard have changed. Before you generate a snapshot, save the dashboard.',
     }),
+  getDraftSharePanelChangesWarning: () =>
+    i18n.translate('dashboard.snapshotShare.panelChangesWarning', {
+      defaultMessage:
+        'You are about to share a dashboard with unsaved changes, and the link may not work properly. Save the dashboard first to create a permanent link.',
+    }),
+  getEmbedShareWarning: () =>
+    i18n.translate('dashboard.embedShare.draftWarning', {
+      defaultMessage:
+        'This dashboard has unsaved changes. Consider saving your dashboard before copying the embed link.',
```

Comment:
We should probably mimic the wording suggested by @florent-leborgne for embedded dashboards, too:

```suggestion
        'You are about to share a dashboard with unsaved changes, and the embed link may not work properly. Save the dashboard first to create a permanent embed link.',
```